### PR TITLE
SDIT-1421 Add builder and tables for basic Incident Report

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/Incident.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    value = [
+      JoinColumnOrFormula(
+        formula = JoinFormula(
+          value = "'" + IncidentType.IR_TYPE + "'",
+          referencedColumnName = "domain",
+        ),
+      ), JoinColumnOrFormula(column = JoinColumn(name = "INCIDENT_TYPE", referencedColumnName = "code")),
+    ],
+  )
+  var type: IncidentType,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/Incident.kt:62]: have you checked in live that there is no dodgy data?

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/Incident.kt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import org.hibernate.annotations.JoinColumnOrFormula
+import org.hibernate.annotations.JoinColumnsOrFormulas
+import org.hibernate.annotations.JoinFormula
+import org.hibernate.type.YesNoConverter
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+
+@Entity
+@Table(name = "INCIDENT_CASES")
```
[the remark this answers is not in the document]
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/Incident.kt:24]: ah yes - I did add to questionnaire after your review - will add in next PR

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/Incident.kt (hunk 1 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/Incident.kt:62]: good point - will double-check.

**Pull request merged**